test/views/honeysql/util_test.clj

```diff
@@ -118,3 +122,95 @@
                :OMM
                :WF}
              (util/query-tables query))))))
+
+(deftest query-tables-delete-test
+  (testing "simple delete"
+    (let [query (hsql/build {:delete-from :foo
+                             :where [:= :id :name]})]
+      (is (= #{:foo} (util/query-tables query))))
+
+    (let [query (hsql/build {:delete-from [:foo :f]
+                             :where [:= :f.id :f.name]})]
+      (is (= #{:foo} (util/query-tables query)))))
+
+  (testing "delete with using"
+    (let [query {:delete-from :foo
+                   :using [:bar]
+                   :where [:= :foo.id :bar.id]}]
+      (is (= #{:foo :bar} (util/query-tables query))))
```

Comment:
A few notes:
- the USING statement can appear in other SQL form including the WHERE portion of a SELECT, but you should have coverage there - AND that is very obscure PSQL syntax. That said, in our codebase, it is only used in the context of DELETE

- Prior to your work, the majority of USING statements in our code worked correctly because the USING clause is drawing from tables defined in a WITH clause. I discovered the error by defining a SELECT inside of the USING clause. ie:
```
...
:using [(hsql/build :select [:*] :from :bar]
...
````

Reply:
More - I will leave it up to you to decide if you have enough coverage - you have certainly done well to address our needs at present :)

Reply:
This test should suffice but could notionally have a case for 
```
    (let [query {:delete-from [:foo :f]
                 :using {{:select [:*] :from :bar} :b}
                 :where [:= :f.id :b.id]}]
      (is (= #{:foo :bar} (util/query-tables query))))
```
which is more explicit form of the issue in https://kirasystems.atlassian.net/browse/DEV-8302